cloudbridge/providers/azure/azure_client.py

```diff
@@ -408,10 +408,10 @@ def delete_vm_firewall_rule(self, fw_rule_id, vm_firewall):
             begin_delete(self.resource_group, vm_firewall, name).result()
 
     def list_containers(self, prefix=None, limit=None, marker=None):
-        results = self.blob_service.list_containers(prefix=prefix,
-                                                    num_results=limit,
+        results = self.blob_service.list_containers(name_starts_with=prefix,
+                                                    results_per_page=limit,
                                                     marker=marker)
-        return (results.items, results.next_marker)
```

Comment:
Azure is returning an [ItemPaged object](https://docs.microsoft.com/en-us/python/api/azure-core/azure.core.paging.itempaged?view=azure-python) which is an iterable, we can return that and let `services.list` do its thing, seems to work okay when I tried. Let me know if you are happy with this.

Out of curiosity how would we pas the prefix to `list_containers`?  if this is called always from `provider.storage.buckets.list()` we can't pass any arguments to it right?